azurerm/resource_arm_mssql_database_vulnerability_assessment_rule_baseline_test.go

```diff
@@ -0,0 +1,276 @@
+  rule_id             = "VA2065"
+  baseline_name       = "master"
+  baseline_result {
+    result = [
+      "clientip",
+      "255.255.255.255",
+      "255.255.255.255"
+    ]
+  }
+
+  depends_on = [
+    azurerm_mssql_server_vulnerability_assessment.test
+  ]
+}
+`, database)
+}
+
+func testAccAzureRMDatabaseVulnerabilityAssessmentRuleBaseline_update(rInt int, location string) string {
+	database := testAccAzureRMMssqlDatabaseVulnerabilityAssessmentRuleBaseline_database(rInt, location)
+
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_mssql_database_vulnerability_assessment_rule_baseline" "test" {
+  resource_group_name = azurerm_resource_group.test.name
+  server_name         = azurerm_sql_server.test.name
+  database_name       = azurerm_sql_database.test.name
+  rule_id             = "VA2065"
+  baseline_name       = "master"
+  baseline_result {
+    result = [
+      "clientips",
+      "255.255.255.255",
+      "255.255.255.255"
+    ]
+  }
+  baseline_result {
+    result = [
+      "myip",
+      "255.255.255.0",
+      "255.255.255.0"
+    ]
+  }
+
+  depends_on = [
+    azurerm_mssql_server_vulnerability_assessment.test
+  ]
+}
+`, database)
+}
+
+func testAccAzureRMMssqlDatabaseVulnerabilityAssessmentRuleBaseline_database(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_sql_server" "test" {
+  name                         = "acctestsqlserver%d"
+  resource_group_name          = azurerm_resource_group.test.name
+  location                     = azurerm_resource_group.test.location
+  version                      = "12.0"
+  administrator_login          = "mradministrator"
+  administrator_login_password = "thisIsDog11"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "accsa%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = "%s"
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+}
+
+resource "azurerm_storage_container" "test" {
+  name                  = "acctestsc%d"
+  storage_account_name  = azurerm_storage_account.test.name
+  container_access_type = "private"
+}
+
+resource "azurerm_mssql_server_security_alert_policy" "test" {
+  resource_group_name = azurerm_resource_group.test.name
+  server_name         = azurerm_sql_server.test.name
+  state               = "Enabled"
+}
+
+resource "azurerm_mssql_server_vulnerability_assessment" "test" {
+  resource_group_name        = azurerm_resource_group.test.name
+  server_name                = azurerm_sql_server.test.name
+  storage_container_path     = "${azurerm_storage_account.test.primary_blob_endpoint}${azurerm_storage_container.test.name}/"
+  storage_account_access_key = azurerm_storage_account.test.primary_access_key
+
+  depends_on = [
+    azurerm_sql_database.test,
+    azurerm_mssql_server_security_alert_policy.test
+  ]
```

Comment:
So this might be a little bit confusing, I already created a support request on Azure side and they confirmed that this is a bug, but they provided no fixing date for now.

* As I explained above the `azurerm_mssql_server_security_alert_policy.test` resource needs to exist before creating `azurerm_mssql_server_vulnerability_assessment` so this is fine
* The dependency to `azurerm_sql_database.test` should actually not be required. You can create the `azurerm_mssql_server_vulnerability_assessment` resource without having the database. The problem was that you needed to run 1 vulnerability scan on the database or do the `azurerm_mssql_server_vulnerability_assessment` setting after creating the database before you are able to set a baseline setting. I just tried it out again and this seems to work now. So I could remove the database dependency here. So the bug seems to be fixed. Shall I remove it?

Reply:
I have removed the database dependency (as well in the documentation), hope this is fine for you.